compiler/modules/types/list.bal

```diff
@@ -236,6 +236,25 @@ function listFormulaIsEmpty(Context cx, Conjunction? pos, Conjunction? neg) retu
     return !listInhabited(cx, indices, memberTypes, nRequired, neg);
 }
 
+function intersectListAtoms(Env env, ListAtomicType[] atoms) returns [SemType, ListAtomicType]? {
+    if atoms.length() == 0 {
+        return ();
+    }
+    ListAtomicType atom = atoms[0];
+    foreach int i in 1 ..< atoms.length() {
+        ListAtomicType next = atoms[i];
+        var intersection = listIntersectWith(atom.members, atom.rest, next.members, next.rest);
```

Comment:
Is this correctly dealing with the case where the intersection of two members is `never`?

Reply:
fixed